core/src/main/java/com/linecorp/armeria/client/ConnectionPoolListener.java

```diff
@@ -50,6 +52,53 @@ static ConnectionPoolListener logging(Ticker ticker) {
         return new ConnectionPoolLoggingListener(ticker);
     }
 
+    /**
+     * Returns a new {@link ConnectionPoolListener} that collects metrics with the specified
+     * {@link MeterRegistry}. The returned listener will generate the following metrics:
+     * <table>
+     * <caption>metrics that will be generated by this class</caption>
+     * <tr>
+     *   <th>metric name</th>
+     *   <th>description</th>
+     * </tr>
+     * <tr>
+     *   <td>{@code armeria.client.connections#count{state="open"}}</td>
+     *   <td>The number of opened connection.</td>
+     * </tr>
+     * <tr>
+     *   <td>{@code armeria.client.connections#count{state="close"}}</td>
+     *   <td>The number of closed connections.</td>
+     * </tr>
+     * </table>
+     */
+    static ConnectionPoolListener metricCollecting(MeterRegistry registry) {
+        return metricCollecting(registry, "armeria.client.connections");
+    }
+
+
+    /**
+     * Returns a new {@link ConnectionPoolListener} that collects metrics with the specified
+     * {@link MeterRegistry} and {@link Meter} name. The returned listener will generate the following metrics:
+     * <table>
+     * <caption>metrics that will be generated by this class</caption>
+     * <tr>
+     *   <th>metric name</th>
+     *   <th>description</th>
+     * </tr>
+     * <tr>
+     *   <td>{@code <name>#count{state="open"}}</td>
+     *   <td>The number of open connection.</td>
+     * </tr>
+     * <tr>
+     *   <td>{@code <name>#count{state="close"}}</td>
+     *   <td>The number of closed connections.</td>
+     * </tr>
+     * </table>
+     */
+    static ConnectionPoolListener metricCollecting(MeterRegistry registry, String name) {
```

Comment:
Should `MeterIdPrefix` as the type of `name`? `MeterIdPrefix` is a commonly used type to represent a metric name in public API.

Reply:
Thanks for your suggestion. Updated.